actors/builtin/power/power_test.go

```diff
@@ -67,6 +68,78 @@ func TestConstruction(t *testing.T) {
 	})
 }
 
+func TestPowerAndPledgeAccounting(t *testing.T) {
+	actor := newHarness(t)
+	owner := tutil.NewIDAddr(t, 101)
+	miner1 := tutil.NewIDAddr(t, 111)
+	miner2 := tutil.NewIDAddr(t, 112)
+
+	// These tests use the min power for consensus to check the accounting above that value.
+	// TODO: tests for crossing the consensus minimum boundary after settling the behaviour.
+	// See https://github.com/filecoin-project/specs-actors/issues/266
+	powerUnit := power.ConsensusMinerMinPower
+
+	mul := func(a big.Int, b int64) big.Int {
+		return big.Mul(a, big.NewInt(b))
+	}
+
+	builder := mock.NewBuilder(context.Background(), builtin.StoragePowerActorAddr).
+		WithCaller(builtin.SystemActorAddr, builtin.SystemActorCodeID)
+
+	t.Run("power & pledge accounted", func(t *testing.T) {
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt)
+
+		actor.createMinerBasic(rt, owner, owner, miner1)
+		actor.createMinerBasic(rt, owner, owner, miner2)
+
+		ret := actor.currentPowerTotal(rt)
+		assert.Equal(t, big.Zero(), ret.RawBytePower)
+		assert.Equal(t, big.Zero(), ret.QualityAdjPower)
+		assert.Equal(t, big.Zero(), ret.PledgeCollateral)
+
+		// Add power for miner1
+		actor.updateClaimedPower(rt, miner1, powerUnit, mul(powerUnit, 2))
+		ret = actor.currentPowerTotal(rt)
+		assert.Equal(t, powerUnit, ret.RawBytePower)
+		assert.Equal(t, mul(powerUnit, 2), ret.QualityAdjPower)
+		assert.Equal(t, big.Zero(), ret.PledgeCollateral)
+
+		// Add power and pledge for miner2
+		actor.updateClaimedPower(rt, miner2, powerUnit, powerUnit)
+		actor.updatePledgeTotal(rt, miner1, abi.NewTokenAmount(1e6))
+		ret = actor.currentPowerTotal(rt)
+		assert.Equal(t, mul(powerUnit, 2), ret.RawBytePower)
+		assert.Equal(t, mul(powerUnit, 3), ret.QualityAdjPower)
+		assert.Equal(t, abi.NewTokenAmount(1e6), ret.PledgeCollateral)
+
+		rt.Verify()
+
+		// Verify claims in state.
+		var st power.State
+		rt.GetState(&st)
+		claim1, found, err := st.GetClaim(rt.AdtStore(), miner1)
+		require.NoError(t, err)
+		require.True(t, found)
+		require.Equal(t, claim1.RawBytePower, powerUnit)
+		require.Equal(t, claim1.QualityAdjPower, mul(powerUnit, 2))
+
+		claim2, found, err := st.GetClaim(rt.AdtStore(), miner2)
+		require.NoError(t, err)
+		require.True(t, found)
+		require.Equal(t, claim2.RawBytePower, powerUnit)
+		require.Equal(t, claim2.QualityAdjPower, powerUnit)
+
+		// Subtract power and some pledge for miner2
+		actor.updateClaimedPower(rt, miner2, powerUnit.Neg(), powerUnit.Neg())
```

Comment:
This test is missing an assertion that miner2's Claim should have zero raw and quality adjusted power after the second update.